### PR TITLE
BUG: DataFrame.unstack with non-consolidated

### LINKED ISSUE #34708: BUG: DataFrame.unstack on non-consolidated frame
```
df = pd.DataFrame({"x": [1, 2, np.NaN], "y": [3.0, 4, np.NaN]})

df2 = df[["x"]]
df2["y"] = df["y"]
assert len(df2._mgr.blocks) == 2

>>> df2.unstack()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pandas/core/frame.py", line 7031, in unstack
    return unstack(self, level, fill_value)
  File "pandas/core/reshape/reshape.py", line 419, in unstack
    return _unstack_frame(obj, level, fill_value=fill_value)
  File "pandas/core/reshape/reshape.py", line 435, in _unstack_frame
    unstacker = _Unstacker(obj.index, level=level)
  File "pandas/core/reshape/reshape.py", line 93, in __init__
    self.index = index.remove_unused_levels()
AttributeError: 'RangeIndex' object has no attribute 'remove_unused_levels'
```

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #34708
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/reshape/reshape.py [hunk 1]
```diff
@@ -415,7 +414,7 @@ def unstack(obj, level, fill_value=None):
         level = obj.index._get_level_number(level)
 
     if isinstance(obj, DataFrame):
-        if isinstance(obj.index, MultiIndex) or not obj._can_fast_transpose:
```
Comment (jreback) [inline pandas/core/reshape/reshape.py:418]: does this have any perf implications?

File: pandas/tests/frame/test_reshape.py [hunk 2]
```diff
@@ -140,6 +140,17 @@ def test_stack_mixed_level(self):
         expected = expected[["a", "b"]]
         tm.assert_frame_equal(result, expected)
 
+    def test_unstack_not_consolidated(self):
+        # Gh#34708
+        df = pd.DataFrame({"x": [1, 2, np.NaN], "y": [3.0, 4, np.NaN]})
+        df2 = df[["x"]]
```
Comment (jreback) [inline pandas/tests/frame/test_reshape.py:146]: don't you need to copy here?

File: pandas/core/reshape/reshape.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/reshape/reshape.py:418]: no, just not-raising implications

File: pandas/tests/frame/test_reshape.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/frame/test_reshape.py:146]: no, we dont care about getting a copy, just non-consolidatedness, which is checked 2 lines down

**Review by jorisvandenbossche — APPROVED** [review]
Looks good to me

**Pull request merged**